### PR TITLE
Tpetra:  Issue3129 makeOptimizedColMapAndImport

### PR DESCRIPTION
**Pull request by kddevin**
@trilinos/tpetra 

## Description
minor changes to address #3129 
consistent return arguments in std::pair
removed unused and/or unneeded arguments

## Motivation and Context
simplify interface

## How Has This Been Tested?
make test on mac, clang compiler

## Checklist

- [ x] My commit messages mention the appropriate GitHub issue numbers.
- [ x] My code follows the code style of the affected package(s).
- [ x] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.
- [ x] No new compiler warnings were introduced.
- [ x] These changes break backwards compatibility.

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp [hunk 1]
```diff
@@ -88,8 +88,7 @@ namespace Details {
     makeOptColMap (std::ostream& errStream,
                    bool& lclErr,
                    const map_type& domMap,
-                   const map_type& colMap,
-                   const import_type* /* oldImport */)
+                   const map_type& colMap)
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp]: It's actually still worthwhile to have an option to pass in the old Import.  The point is that we could use the information in the old Import object, if provided, to construct the optimized column Map with less communication.  (For example, the old Import knows what GIDs in the input column Map are local vs. remote, and it knows the remote GIDs' owning PIDs.)  We don't implement that optimization now, but it wouldn't be too hard to add later, and the current interface permits that.

File: packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp [hunk 2]
```diff
@@ -335,8 +325,7 @@ namespace Details {
     makeOptColMapAndImport (std::ostream& errStream,
                             bool& lclErr,
                             const map_type& domMap,
-                            const map_type& colMap,
-                            const import_type* oldImport)
+                            const map_type& colMap)
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp]: See above comment.

File: packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp [hunk 3]
```diff
@@ -378,24 +367,19 @@ namespace Details {
...
+  Teuchos::RCP<const MapType>
   makeOptimizedColMap (std::ostream& errStream,
                        bool& lclErr,
                        const MapType& domMap,
-                       const MapType& colMap,
-                       const Tpetra::Import<
-                         typename MapType::local_ordinal_type,
-                         typename MapType::global_ordinal_type,
-                         typename MapType::node_type>* oldImport = nullptr)
+                       const MapType& colMap)
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp]: See above comment.

File: packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp [hunk 4]
```diff
@@ -453,28 +434,21 @@ namespace Details {
...
-  std::pair<MapType, Teuchos::RCP<typename OptColMap<MapType>::import_type> >
+  std::pair<Teuchos::RCP<const MapType>,
+            Teuchos::RCP<typename OptColMap<MapType>::import_type> >
   makeOptimizedColMapAndImport (std::ostream& errStream,
                                 bool& lclErr,
                                 const MapType& domMap,
-                                const MapType& colMap,
-                                const typename OptColMap<MapType>::import_type* oldImport,
-                                const bool makeImport)
+                                const MapType& colMap)
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp]: See above comment.  It makes sense to keep the optional `oldImport` argument.

File: packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp [hunk 5]
```diff
@@ -378,24 +367,19 @@ namespace Details {
   /// It does everything that that function does, except that it does
   /// not compute a new Import.
   template<class MapType>
-  MapType
+  Teuchos::RCP<const MapType>
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_Details_makeOptimizedColMap.hpp]: It' doesn't really matter whether we return Map by const or nonconst RCP.  We could get rid of const here.  We may want to take a vote on that.

**Comment by kddevin** [comment]
Thanks for the review, @mhoemmen 
If we keep oldImport, we need to be more consistent about it.

For instance, it wasn't documented for the recommended helper functions makeOptimizedColMap and makeOptimizedColMapAndImport.
https://trilinos.org/docs/dev/packages/tpetra/doc/html/namespaceTpetra_1_1Details.html#a7c72e02868dd5193c1d36fe416a7224e

For the class functions, the documentation said it would accessed only if makeImport=true; thus, passing it to makeOptColMap without passing makeImport was confusing.  Would it be helpful in makeOptColMap or just in recreating the new Import?

**Comment by kddevin** [comment]
OK, oldImport is restored. Please re-review.  Thanks.

**Pull request merged**